### PR TITLE
Add r-vctrs

### PR DESCRIPTION
**Pull request by anutosh491**
Depends on #1303

### CODE REVIEW

**Comment by anutosh491** [comment]
Hmm i get this error 
```
 × error └─ r-lifecycle 1.0.4 | 1.0.4 | 1.0.4 | 1.0.4 would require
 × error    └─ r-base >=4.3,<4.4.0a0, for which no candidates were found.
Error:   × Failed to resolve dependencies: Cannot solve the request because of: r-
  │ lifecycle >=1.0.4 cannot be installed because there are no viable options:
  │ └─ r-lifecycle 1.0.4 | 1.0.4 | 1.0.4 | 1.0.4 would require
  │    └─ r-base >=4.3,<4.4.0a0, for which no candidates were found.
  │ 
  ╰─▶ Cannot solve the request because of: r-lifecycle >=1.0.4 cannot be
      installed because there are no viable options:
      └─ r-lifecycle 1.0.4 | 1.0.4 | 1.0.4 | 1.0.4 would require
         └─ r-base >=4.3,<4.4.0a0, for which no candidates were found.
         
```
I guess r-lifecycle is built with the latest r-base (https://github.com/conda-forge/r-lifecycle-feedstock/pull/11)

**Comment by IsabelParedes** [comment]
This doesn't make sense...
```
╰─λ micromamba repoquery depends r-lifecycle
Using local repodata...

Name        Version Build           Channel     Subdir
──────────────────────────────────────────────────────────
r-base      4.4.1   h745b8e0_14     conda-forge linux-64
r-cli       3.6.3   r44h0d4f4ea_1   conda-forge linux-64
r-glue      1.7.0   r44hb1dbf0f_1   conda-forge linux-64
r-lifecycle 1.0.4   r44hc72bb7e_1   conda-forge noarch
r-rlang     1.1.4   r44ha18555a_1   conda-forge linux-64
```

File: recipes/recipes_emscripten/r-vctrs/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - ${{ compiler('cxx') }}
+  host:
+  - r-base == ${{ r_base_version }}
+  - r-cli >=3.6.3
+  - r-glue
+  - r-rlang
+  - r-lifecycle >=1.0.4
+  run:
+  - r-base == ${{ r_base_version }}
+  - r-cli >=3.6.3
```
Comment (IsabelParedes) [inline recipes/recipes_emscripten/r-vctrs/recipe.yaml:40]: Good catch. It looks like we need another recipe for `r-cli` because only v3.0.1 is noarch :(

Comment (anutosh491) [reply recipes/recipes_emscripten/r-vctrs/recipe.yaml:40]: Yess, I got everything to build locally. So r-cli v3.0.1 is noarch so technically we can either use r-cli <3 with r-base 4.1 or we have to build latest r-cli for ourselves with r-base 4.4.1. This was pointed out by the recipe maintainers here too(https://github.com/conda-forge/r-cli-feedstock/issues/20#issuecomment-888607052)

This is how I got everything to work locally 
1) build r-cli latest
2) Use that r-cli compatible with R 4.4.1 for r-lifecycle 
3) Then r-vctrs, r-purrr, r-tibble (dependening on r-cli and r-lifecycle) was straightforward.

**Comment by anutosh491** [comment]
This should be ready, just needs to be rebased on top of #1364 and #1365

**Comment by anutosh491** [comment]
Okay wasn't expecting this but worked right off the bat. Thanks for the reviews @IsabelParedes :)

**Pull request merged**